test/11_array.jl

```diff
@@ -1,635 +1,639 @@
+    @test isordered(x2) == isordered(x)
+
+    x2 = x[1:1]
+    @test typeof(x2) === typeof(x)
+    @test x2 == ["b"]
+    @test levels(x2) == levels(x)
+    @test levels(x2) !== levels(x)
+    @test isordered(x2) == isordered(x)
+
+    x2 = x[2:1]
+    @test typeof(x2) === typeof(x)
+    @test isempty(x2)
+    @test levels(x2) == levels(x)
+    @test levels(x2) !== levels(x)
+    @test isordered(x2) == isordered(x)
+
+    x[1] = x[2]
+    @test x[1] === x.pool.valindex[2]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[1]
+
+    x[3] = "c"
+    @test x[1] === x.pool.valindex[2]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[3]
+    @test levels(x) == ["a", "b", "c"]
+
+    x[2:3] = "b"
+    @test x[1] === x.pool.valindex[2]
+    @test x[2] === x.pool.valindex[1]
+    @test x[3] === x.pool.valindex[1]
+    @test levels(x) == ["a", "b", "c"]
+
+    @test droplevels!(x) === x
+    @test levels(x) == ["a", "b"]
+    @test x[1] === x.pool.valindex[1]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[2]
+    @test levels(x) == ["a", "b"]
+
+    @test levels!(x, ["b", "a"]) === x
+    @test levels(x) == ["b", "a"]
+    @test x[1] === x.pool.valindex[1]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[2]
+    @test levels(x) == ["b", "a"]
+
+    @test_throws ArgumentError levels!(x, ["a"])
+    @test_throws ArgumentError levels!(x, ["e", "b"])
+    @test_throws ArgumentError levels!(x, ["e", "a", "b", "a"])
+
+    @test levels!(x, ["e", "a", "b"]) === x
+    @test levels(x) == ["e", "a", "b"]
+    @test x[1] === x.pool.valindex[1]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[2]
+    @test levels(x) == ["e", "a", "b"]
+
+    x[1] = "c"
+    @test x[1] === x.pool.valindex[4]
+    @test x[2] === x.pool.valindex[2]
+    @test x[3] === x.pool.valindex[2]
+    @test levels(x) == ["e", "a", "b", "c"]
+
+    push!(x, "a")
+    @test length(x) == 4
+    @test x[end] == "a"
+    @test levels(x) == ["e", "a", "b", "c"]
+
+    push!(x, "zz")
+    @test length(x) == 5
+    @test x[end] == "zz"
+    @test levels(x) == ["e", "a", "b", "c", "zz"]
+
+    push!(x, x[1])
+    @test length(x) == 6
+    @test x[1] == x[end]
+    @test levels(x) == ["e", "a", "b", "c", "zz"]
+
+    append!(x, x)
+    @test length(x) == 12
+    @test x == ["c", "b", "b", "a", "zz", "c", "c", "b", "b", "a", "zz", "c"]
+    @test isordered(x) === false
+    @test levels(x) == ["e", "a", "b", "c", "zz"]
+
+    b = ["z","y","x"]
+    y = CategoricalVector{String, R}(b)
+    append!(x, y)
+    @test length(x) == 15
+    @test x == ["c", "b", "b", "a", "zz", "c", "c", "b", "b", "a", "zz", "c", "z", "y", "x"]
+    @test levels(x) == ["e", "a", "b", "c", "zz", "x", "y", "z"]
+
+    empty!(x)
+    @test length(x) == 0
+    @test levels(x) == ["e", "a", "b", "c", "zz", "x", "y", "z"]
+
+    # Vector created from range (i.e. non-Array AbstractArray),
```

Comment:
Should be a testset for consistency with 12_nullablearray.jl. More generally, please check that all changes are applied in the same way to both files when applicable, since they should be kept in sync.